appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java

```diff
@@ -16,61 +16,262 @@
+    private final SpreadsheetService spreadsheetService;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public ExamResource(ExamService examService, TokenService tokenService, UserTokenDao userTokenDao, ExamMetadataDao examMetadataDao, QuestionsDao questionsDao, ExamInstanceDao examInstanceDao, AnswerDao answerDao, SpreadsheetService spreadsheetService, ObjectMapper objectMapper) {
+        this.examService = examService;
+        this.tokenService = tokenService;
+        this.userTokenDao = userTokenDao;
+        this.examMetadataDao = examMetadataDao;
+        this.questionsDao = questionsDao;
+        this.examInstanceDao = examInstanceDao;
+        this.answerDao = answerDao;
+        this.spreadsheetService = spreadsheetService;
+        this.objectMapper = objectMapper;
+    }
+
     /**
-     * Get corresponding userID from the IDToken using tokenVerifier
-     * Get tokens for the user from the database
-     * Use the tokens to read exam from the spreadsheet
-     * Validate exam
-     * Convert exam from List<List<Object>> to Exam object
+     * Called when paper setter submits question paper
+     * Gets corresponding userID from the IDToken using tokenVerifier
+     * Get tokens for the user from the database and use the tokens to read exam from the spreadsheet
+     * Validate exam and convert exam from List<List<Object>> to Exam object
      * post examMetadata in database to get examID
+     * make response template in google sheets for this exam
      * post exam into the database
      *
-     * @param IDToken ( from header )
-     * @param request ( must contain spreadsheetID, sheetName )
+     * @param IDToken ( used to authenticate user, here paper setter )
+     * @param request ( contains spreadsheetID, sheetName )
      * @return Exam object
+     * @throws BadRequestException          ( if IDToken is invalid,
+     *                                      if user cannot access spreadsheet given in the request,
+     *                                      if the question paper is not following correct format as defined in template )
+     * @throws InternalServerErrorException ( by GeneralSecurityException and IOException for credentials file,
+     *                                      if the database contains malformed used token data,
+     *                                      unable to create response sheet in spreadsheet for this exam )
+     * @throws NotAuthorizedException       ( if the user is not authorized i.e. no tokens for user available in database )
      */
     @POST
-    public ExamResponse postExam(@HeaderParam("authorization-code") String IDToken, ExamRequest request) {
-        return null;
+    public ExamResponse postExam(@NotNull @HeaderParam("Authentication") String IDToken, @NotNull ExamRequest request) {
+        String userID;
+        User token;
+        Exam exam;
+        try {
+            userID = tokenService.verifyIDToken(IDToken);
```

Comment:
A lot of duplicate code, can you please all the try catch to the service itself. This method will be throwing all the server related exceptions.

Reply:
Added a TODO for this.